packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -666,11 +666,49 @@ export function makeVatKeeper(
     });
   }
 
-  function deleteSnapshotsAndTranscript() {
-    if (snapStore) {
-      snapStore.deleteVatSnapshots(vatID);
+  /**
+   * Perform some (possibly-limited) cleanup work for a vat. Returns
+   * 'done' (where false means "please call me again", and true means
+   * "you can delete the vatID now"), and a count of how much work was
+   * done (so the runPolicy can decide when to stop).
+   *
+   * @param {number} [budget]
+   * @returns {{ done: boolean, cleanups: number }}
+   *
+   */
+  function deleteSnapshots(budget = undefined) {
+    // Each budget=1 allows us to delete one snapshot entry.
+    let cleanups = 0;
+    if (!snapStore) {
+      return { done: true, cleanups };
     }
-    transcriptStore.deleteVatTranscripts(vatID);
+    // initially uses 2+2*budget DB statements, then just 1 when done
+    const dc = snapStore.deleteVatSnapshots(vatID, budget);
+    cleanups = dc.cleanups;
+    return { done: dc.done, cleanups };
```

Comment:
Why not just return directly?
```suggestion
    const dc = snapStore.deleteVatSnapshots(vatID, budget);
    return dc;
```
(and likewise for `deleteTranscripts`)
or
```suggestion
    return snapStore.deleteVatSnapshots(vatID, budget);
```

Reply:
oh, yeah, good idea, that's a leftover of when I was deleting both snapshots and transcripts in the same function